### PR TITLE
Replace hard coded dubbo-config-api values and small code optimization

### PR DESCRIPTION
**Pull request by khanimteyaz**
In this PR I have refactored code to

1.  Replace hard coded values from java code to constant. Most of this values where getting used from multiple places so moving to constant avoid development time or run time error for value mismatch.
2.  Small refactoring to reduce code volume and use existing JDK provided utility and operation.

 Bellow are the java file which has been modified as part of this PR

  dubbo-common/src/main/java/org/apache/dubbo/common/Constants.java
  dubbo-common/src/main/java/org/apache/dubbo/common/EnvironmentConstants.java
  dubbo-config/dubbo-config-api/src/main/java/org/apache/dubbo/config/AbstractServiceConfig.java
  dubbo-config/dubbo-config-api/src/main/java/org/apache/dubbo/config/ApplicationConfig.java
  dubbo-config/dubbo-config-api/src/main/java/org/apache/dubbo/config/ConfigCenterConfig.java
  dubbo-config/dubbo-config-api/src/main/java/org/apache/dubbo/config/MethodConfig.java
  dubbo-config/dubbo-config-api/src/main/java/org/apache/dubbo/config/ModuleConfig.java
  dubbo-config/dubbo-config-api/src/main/java/org/apache/dubbo/config/ProtocolConfig.java
  dubbo-config/dubbo-config-api/src/main/java/org/apache/dubbo/config/ProviderConfig.java
  dubbo-config/dubbo-config-api/src/main/java/org/apache/dubbo/config/ReferenceConfig.java
  dubbo-config/dubbo-config-api/src/main/java/org/apache/dubbo/config/RegistryConfig.java
  dubbo-config/dubbo-config-api/src/main/java/org/apache/dubbo/config/ServiceConfig.java

## What is the purpose of the change

1.  Replace hard coded values from java code to constant. Most of this values where getting used from multiple places so moving to constant avoid development time or run time error for value mismatch.
2.  Small refactoring to reduce code volume and use existing JDK provided utility and operation.

## Brief changelog

Hard coded value replacement of dubbo-config-api and small refactoring to optimize code size.

## Verifying this change

Running UT

Follow this checklist to help us incorporate your contribution quickly and easily:

- [x] Make sure there is a [3107](https://github.com/apache/incubator-dubbo/issues/3107) field for the change (usually before you start working on it). Trivial changes like typos do not require a GITHUB issue. Your pull request should address just this issue, without pulling in other changes - one PR resolves one issue.
- [ ] Format the pull request title like `[Dubbo-XXX] Fix UnknownException when host config not exist #XXX`. Each commit in the pull request should have a meaningful subject line and body.
- [ ] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
- [ ] Write necessary unit-test to verify your logic correction, more mock a little better when cross module dependency exist. If the new feature or significant change is committed, please remember to add integration-test in [test module](https://github.com/apache/incubator-dubbo/tree/master/dubbo-test).
- [ ] Run `mvn clean install -DskipTests=false` & `mvn clean test-compile failsafe:integration-test` to make sure unit-test and integration-test pass.
- [ ] If this contribution is large, please follow the [Software Donation Guide](https://github.com/apache/incubator-dubbo/wiki/Software-donation-guide).

### CODE REVIEW

File: dubbo-common/src/main/java/org/apache/dubbo/common/EnvironmentConstants.java [hunk 1]
```diff
@@ -0,0 +1,60 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
```
Comment (beiwei30) [inline dubbo-common/src/main/java/org/apache/dubbo/common/EnvironmentConstants.java]: suggest to put all constants in Constants, and make it an interface. After that, let's revisit this class to consider if it is possible to split. What do you think?

File: dubbo-config/dubbo-config-api/src/main/java/org/apache/dubbo/config/ApplicationConfig.java [hunk 2]
```diff
@@ -150,10 +151,18 @@ public String getEnvironment() {
...
+        checkName(EnvironmentConstants.ENVIRONMENT, environment);
         if (environment != null) {
-            if (!("develop".equals(environment) || "test".equals(environment) || "product".equals(environment))) {
-                throw new IllegalStateException("Unsupported environment: " + environment + ", only support develop/test/product, default is product.");
+            if (!(EnvironmentConstants.DEVELOPMENT_ENVIRONMENT.equals(environment)
+                    || EnvironmentConstants.TEST_ENVIRONMENT.equals(environment)
+                    || EnvironmentConstants.PRODUCTION_ENVIRONMENT.equals(environment))) {
+
+                throw new IllegalStateException(String.format("Unsupported environment: %s, only support %s/%s/%s, default is %s.",
+                         environment
```
Comment (code4wt) [inline dubbo-config/dubbo-config-api/src/main/java/org/apache/dubbo/config/ApplicationConfig.java]: Hi, this code style looks a bit strange, I suggest writing the comma after the variable, like this:

```java
throw new IllegalStateException(String.format("Unsupported environment: %s, only support %s/%s/%s, default is %s.",
        environment,
        EnvironmentConstants.DEVELOPMENT_ENVIRONMENT,
        EnvironmentConstants.TEST_ENVIRONMENT,
        EnvironmentConstants.PRODUCTION_ENVIRONMENT,
        EnvironmentConstants.PRODUCTION_ENVIRONMENT));
```

What do you think?

Comment (khanimteyaz) [reply dubbo-config/dubbo-config-api/src/main/java/org/apache/dubbo/config/ApplicationConfig.java]: sure.

File: dubbo-common/src/main/java/org/apache/dubbo/common/EnvironmentConstants.java (hunk 1 above)
Comment (khanimteyaz) [reply dubbo-common/src/main/java/org/apache/dubbo/common/EnvironmentConstants.java]: On it. One quick check, if I change the Constants to an interface and dubbo ecosystem projects(outside of incubator-dubbo) are using this class by extending then they might start failing. Do you know any one of them using it? If no one is using then we can change it to interface now.

Comment (khanimteyaz) [reply dubbo-common/src/main/java/org/apache/dubbo/common/EnvironmentConstants.java]: Leaving the changing of java class to interface change. Or if you feel we should go ahead and make change please let the comment I will make the changes as part of this PR only.

**Pull request merged**